lib/Routing/RedirectHandler.php

```diff
@@ -228,6 +232,7 @@ protected function buildRedirectResponse(Redirect $redirect, Request $request, $
         }
 
         $statusCode = $redirect->getStatusCode() ?: Response::HTTP_MOVED_PERMANENTLY;
+        $this->dispatchEvent(new RedirectEvent($redirect), RedirectEvents::PRE_BUILD);
```

Comment:
I would move dispatch before `$url` and `$statusCode` generation so that updated info is used in Response.

Reply:
@alyna